### PR TITLE
Separate active vs completed Cosmos proposal queries

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #3152

## Description of Changes
- Lazy-loads completed Cosmos proposals after chain init
- New directory structure for explicit separation of `v1` vs `v1beta1`
- Move some controller functions to `utils` files for easier testing and reusability 

## Test Plan
- CA (click around) tested on local:
  - http://localhost:8080/osmosis/proposals (v1beta1)
  - http://localhost:8080/kyve/proposals (v1)
  - Query tests for v1 and v1beta1 getCompletedProposals and getActiveProposals
  - update proposals.spec e2e test to account for lazy-loading

## Other Considerations

- Adds a spinner to the lazy-loading proposals. We can leave this space blank if preferred.
- https://www.loom.com/share/c49b253f0ec74dd6ae33a88c9bad5457

### CODE REVIEW

**Comment by jnaviask** [comment]
@mhagel What's left for this PR? Would love for it to go out in the next few days!

File: packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx
Comment (mhagel) [inline packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx:1]: This is the hook that lazy-loads completed proposals on the /proposals page

**Comment by mhagel** [comment]
@jnaviask got tests sorted out. R4R now

I am leaving some v1beta1 tests commented out, to be addressed in #3563 
They work locally for osmosis.

**Comment by mhagel** [comment]
Sorry for the large diff. Most of this is moving methods and files around. I will highlight important changes.

File: packages/commonwealth/client/scripts/views/pages/proposals.tsx [hunk 1]
```diff
@@ -46,6 +48,10 @@ function getModules(): ProposalModule<any, any, any>[] {
 const ProposalsPage = () => {
   const [isLoading, setLoading] = useState(!app.chain || !app.chain.loaded);
   const [isSubstrateLoading, setSubstrateLoading] = useState(false);
+  const {
+    completedCosmosProposals,
+    isLoading: isCosmosCompletedProposalsLoading,
+  } = useGetCompletedCosmosProposals({ app });
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/views/pages/proposals.tsx:55]: here is the lazy load

File: packages/commonwealth/test/integration/api/devnet/cosmos/proposalTx.spec.ts [hunk 2]
```diff
@@ -186,47 +170,134 @@ describe('Proposal Tests', () => {
...
+            ]);
+          });
+        });
+      });
     });
   });
 });
+
+// TODO: use a v1beta1 devnet for CI testing. See: https://github.com/hicommonwealth/commonwealth/issues/3563
+
```
Comment (mhagel) [inline packages/commonwealth/test/integration/api/devnet/cosmos/proposalTx.spec.ts:259]: these work locally, but not in CI. I will convert them to a v1beta1 devnet in the next issue, as noted

**Review by jnaviask — CHANGES_REQUESTED** [review]
Getting infinite bobber on /proposals page for both osmosis and juno when I navigate via the sidebar after loading the forum. Works fine when I navigate directly to the proposals page.

**Comment by mhagel** [comment]
>Getting infinite bobber on /proposals page for both osmosis and juno when I navigate via the sidebar after loading the forum. Works fine when I navigate directly to the proposals page.

Was an existing issue. Fixed.
@jnaviask

**Comment by jnaviask** [comment]
Great -- looks great, however, it's reloading the proposals every time you navigate to the page during a session, rather than simply pulling the data out from the store. We should really not fetch this data more than once per visit.

File: packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const [isLoading, setLoading] = useState(false);
+  const [completedCosmosProposals, setCompletedCosmosProposals] = useState<
+    CosmosProposal[]
+  >([]);
+  useEffect(() => {
+    const getProposals = async () => {
+      const cosmos = app.chain as Cosmos;
+      const storedProposals =
+        cosmos.governance.store.getAll() as CosmosProposal[];
+
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx:28]: @jnaviask now using the store instead of refetching

Comment (mhagel) [reply packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx:28]: wait, not ready yet

**Review by mhagel — COMMENTED** [review]
still need to fix something

File: packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx (hunk 3 above)
Comment (mhagel) [reply packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx]: ready now

Comment (mhagel) [reply packages/commonwealth/client/scripts/hooks/cosmos/useGetCompletedCosmosProposals.tsx]: Fixed duplicate call issue with useRef

**Comment by mhagel** [comment]
>still need to fix something
all fixed and ready to go!

**Review by jnaviask — APPROVED** [review]

**Pull request merged**